example/exceptions_tests.cpp

```diff
@@ -89,4 +97,9 @@ void ExceptionTests()
     AssertThrows(AssertionException, LastException<std::logic_error>());
     AssertThat(LastException<AssertionException>().GetMessage(), Contains("No exception was stored"));
   }
+
+  it("prints description of unwanted exception");
+  {
+    AssertTestFails(AssertThrows(ExceptionWeExcepted, objectUnderTest.LogicError()), "Expected ExceptionWeExcepted. Wrong exception was thrown. Description of unwated exception: not logical!");
+  }
 }
```

Comment:
Speaking of commit messages: The commit subjects (first line) are way too long. Even GitHub does not display it in full length.

For example, change them to, for example:
* Print description of unexpected exception in AssertThrows
* Test printing unexpected exception in AssertThrows